examples/contentful-simple/app/blog/[slug]/page.tsx

```diff
@@ -0,0 +1,49 @@
+import { notFound } from 'next/navigation'
+
+import { MakeswiftComponent } from '@makeswift/runtime/next'
+import { getSiteVersion } from '@makeswift/runtime/next/server'
+
+import { BLOG_CONTENT_WITH_SLOT_TYPE } from '@/components/BlogContentWithSlot/BlogContentWithSlot.makeswift'
+import { GetBlogsDocument, GetBlogsQuery } from '@/generated/contentful'
+import { client } from '@/lib/contentful/client'
+import { getAllBlogs } from '@/lib/contentful/fetchers'
+import { ContentfulProvider } from '@/lib/contentful/provider'
+import { client as MakeswiftClient } from '@/lib/makeswift/client'
+
+export async function generateStaticParams() {
+  const blogs = await getAllBlogs()
+  return blogs.map(blog => ({ slug: blog?.slug }))
+}
+
+export default async function Page({ params }: { params: Promise<{ slug: string }> }) {
+  const { slug } = await params
+  if (!slug) {
+    return notFound()
+  }
+
+  const componentSnapshot = await MakeswiftClient.getComponentSnapshot(
```

Comment:
We shouldn't use `getComponentSnapshot` directly in application-level code. Our convention is to create a [user-land wrapper](https://github.com/makeswift/makeswift/blob/main/examples/contentful/lib/makeswift/component.tsx) that combines `getComponentSnapshot` and `<MakeswiftComponent>` into a friendlier higher-level construct.